pkgs/by-name/ru/rustdesk/package.nix

```
@@ -33,50 +33,61 @@
+      "keepawake-0.4.3" = "sha256-cqSpkq/PCz+5+ZUyPy5hF6rP3fBzuZDywyxMUQ50Rk4=";
       "machine-uid-0.3.0" = "sha256-rEOyNThg6p5oqE9URnxSkPtzyW8D4zKzLi9pAnzTElE=";
       "magnum-opus-0.4.0" = "sha256-T4qaYOl8lCK1h9jWa9KqGvnVfDViT9Ob5R+YgnSw2tg=";
-      "mouce-0.2.1" = "sha256-3PtNEmVMXgqKV4r3KiKTkk4oyCt4BKynniJREE+RyFk=";
-      "pam-0.7.0" = "sha256-qe2GH6sfGEUnqLiQucYLB5rD/GyAaVtm9pAxWRb1H3Q=";
-      "parity-tokio-ipc-0.7.3-2" = "sha256-WXDKcDBaJuq4K9gjzOKMozePOFiVX0EqYAFamAz/Yvw=";
-      "rdev-0.5.0-2" = "sha256-MJ4Uqp0yz1CcFvoZYyUYwNojUcfW1AyVowKShihhhbY=";
-      "reqwest-0.11.18" = "sha256-3k2wcVD+DzJEdP/+8BqP9qz3tgEWcbWZj5/CjrZz5LY=";
+      "pam-0.7.0" = "sha256-o47tVoFlW9RiL7O8Lvuwz7rMYQHO+5TG27XxkAdHEOE=";
+      "pam-sys-1.0.0-alpha4" = "sha256-5HIErVWnanLo5054NgU+DEKC2wwyiJ8AHvbx0BGbyWo=";
+      "parity-tokio-ipc-0.7.3-4" = "sha256-PKw2Twd2ap+tRrQxqg8T1FvpoeKn0hvBqn1Z44F1LcY=";
+      "rdev-0.5.0-2" = "sha256-KrzNa4sKyuVw3EV/Ec9VBNRyJy7QFR2Gu4c2WkltwUw=";
+      "reqwest-0.11.23" = "sha256-kEUT+gs4ziknDiGdPMLnj5pmxC5SBpLopZ8jZ34GDWc=";
       "rust-pulsectl-0.2.12" = "sha256-8jXTspWvjONFcvw9/Z8C43g4BuGZ3rsG32tvLMQbtbM=";
-      "sciter-rs-0.5.57" = "sha256-NQPDlMQ0sGY8c9lBMlplT82sNjbgJy2m/+REnF3fz8M=";
-      "tao-0.22.2" = "sha256-vZx7WM6vK9UarbFQ/FMnTNEEDS+tglhWcPXt/h7YMFA=";
-      "tfc-0.6.1" = "sha256-ukxJl7Z+pUXCjvTsG5Q0RiXocPERWGsnAyh3SIWm0HU=";
-      "tokio-socks-0.5.1-2" = "sha256-x3aFJKo0XLaCGkZLtG9GYA+A/cGGedVZ8gOztWiYVUY=";
-      "tray-icon-0.5.1" = "sha256-1VyUg8V4omgdRIYyXhfn8kUvhV5ef6D2cr2Djz2uQyc=";
+      "sciter-rs-0.5.57" = "sha256-5Nd9npdx8yQJEczHv7WmSmrE1lBfvp5z7BubTbYBg3E=";
+      "sysinfo-0.29.10" = "sha256-/UsFAvlWs/F7X1xT+97Fx+pnpCguoPHU3hTynqYMEs4=";
+      "tao-0.25.0" = "sha256-kLmx1z9Ybn/hDt2OcszEjtZytQIE+NKTIn9zNr9oEQk=";
+      "tfc-0.7.0" = "sha256-VAoOadgGajWjW7q1RCwmQ5gMmaA1oeZBSiL1tYSuIEI=";
+      "tokio-socks-0.5.2-1" = "sha256-i1dfNatqN4dinMcyAdLhj9hJWVsT10OWpCXsxl7pifI=";
+      "tray-icon-0.15.1" = "sha256-DvTVAsE3dB+o89SFk3RPRt5JzaWjPl4nlvC77zsVN/s=";
+      "wallpaper-3.2.0" = "sha256-p9NRmusdA0wvF6onp1UTL0/4t7XnEAc19sqyGDnfg/Q=";
+      "webm-1.1.0" = "sha256-p4BMej7yvb8c/dJynRWZmwo2hxAAY96Qx6Qx2DbT8hE=";
       "x11-2.19.0" = "sha256-GDCeKzUtvaLeBDmPQdyr499EjEfT6y4diBMzZVEptzc=";
+      "x11-clipboard-0.8.1" = "sha256-PtqmSD2MwkbLVWbfTSXZW3WEvEnUlo04qieUTjN2whE=";
     };
   };
 
-  prePatch = ''
-    # Patch 404 repos
-    substituteInPlace Cargo.lock --replace-fail "fufesou" "rustdesk-org";
-    substituteInPlace Cargo.toml --replace-fail "fufesou" "rustdesk-org";
-    substituteInPlace libs/enigo/Cargo.toml --replace-fail "fufesou" "rustdesk-org";
-  '';
+  patches = [
+    # Currently, cargo.lock is broken, so we need to update it using a patch
+    (fetchpatch {
+      url = "https://github.com/rustdesk/rustdesk/pull/9171/commits/6a2bc35e542cdde5944501e1ef88b4005e7b3991.patch";
+      hash = "sha256-/MGHuk7icB8frSbyKsZsjBYXp+IyngdHR5pZu3SGSO8=";
+    })
+  ];
```

Why is this needed if Cargo.lock is already vendored? Any patches required for it should be applied ahead of time. Per [this](https://github.com/NixOS/nixpkgs/blob/aabb78cb7ade02af5b8d4b07b534531972a91c0f/doc/languages-frameworks/rust.section.md#importing-a-cargolock-file-importing-a-cargolock-file) section in our Rust documentation, you should probably be doing this instead

```suggestion
  postPatch = ''
    ln -s ${./Cargo.lock} Cargo.lock
  '';
```

---

Ok yes this is way better with a ln.
without it, I get this error : 
```bash
building '/nix/store/yh8s0widi99xihyf7igz27rwch2pq9vn-cargo-vendor-dir.drv'...
ln: failed to create symbolic link '/nix/store/8hapv8ax8i7631is0wfsrhq93vkcsxbj-cargo-vendor-dir/core-foundation-sys-0.8.6/8ryg75d8l1wwd7aw5s4zy4qywgvkw0l3-core-foundation-sys-0.8.6': Permission denied
error: builder for '/nix/store/yh8s0widi99xihyf7igz27rwch2pq9vn-cargo-vendor-dir.drv' failed with exit code 1;
       last 1 log lines:
       > ln: failed to create symbolic link '/nix/store/8hapv8ax8i7631is0wfsrhq93vkcsxbj-cargo-vendor-dir/core-foundation-sys-0.8.6/8ryg75d8l1wwd7aw5s4zy4qywgvkw0l3-core-foundation-sys-0.8.6': Permission denied
       For full logs, run 'nix log /nix/store/yh8s0widi99xihyf7igz27rwch2pq9vn-cargo-vendor-dir.drv'.
error: 1 dependencies of derivation '/nix/store/9fw71vs9gvdjwd771ir2nnly8ddybdj0-rustdesk-1.3.0.drv' failed to build
```

---

This is probably caused by the Cargo.lock having two entires for `core-foundation-sys` as shown [here](https://github.com/NixOS/nixpkgs/blob/271ffd08a7c5cf41e9d367bbbf9e549fd166868c/pkgs/by-name/ru/rustdesk/Cargo.lock#L1234-L1246). See https://github.com/NixOS/nixpkgs/issues/183344